JFRWebSocket.m

```diff
@@ -717,6 +725,46 @@ -(void)dealloc
 
 /////////////////////////////////////////////////////////////////////////////
 /////////////////////////////////////////////////////////////////////////////
+@interface JFRThread ()
+
+@property(nonatomic, assign)BOOL isRunning;
+@property(nonatomic, strong)NSRunLoop *loop;
+
+@end
+/////////////////////////////////////////////////////////////////////////////
+@implementation JFRThread
+
+/////////////////////////////////////////////////////////////////////////////
++(NSRunLoop*)sharedLoop {
+    return [self shared].loop;
+}
+/////////////////////////////////////////////////////////////////////////////
++(JFRThread*)shared {
```

Comment:
Use `instancetype` as return type instead. Idiomatic and allows subclassing in the future.